### PR TITLE
Fix clippy lints 1.83, bump deps

### PR DESCRIPTION
**Pull request by nyurik**

### CODE REVIEW

File: martin/src/pg/utils.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ use tokio::time::timeout;
 use crate::source::UrlQuery;
 
 #[cfg(test)]
+#[expect(clippy::ref_option)]
```
Comment (CommanderStorm) [inline martin/src/pg/utils.rs]: CI is failing due to it not knowing this lint.

image

=> would it be an option to postpone this until CI is updated?

```suggestion
```

Comment (nyurik) [reply martin/src/pg/utils.rs]: Yep, totally not in a hurry:)

Comment (nyurik) [reply martin/src/pg/utils.rs]: Ps ironically, I'm the one who created that lint :)

**Review by CommanderStorm — APPROVED** [review]
LGTM

**Pull request merged**